api/test/wfl/integration/modules/aou_test.clj

```diff
@@ -46,21 +46,64 @@
            (is (s/valid? :wfl.api.spec/append-to-aou-response response))
            (is (empty? response)))))))
 
-(deftest test-update-unstarted
-  (let [workload (->> (make-aou-workload-request)
-                      workloads/create-workload!
-                      workloads/update-workload!)]
-    (is (nil? (:finished workload)))
-    (is (nil? (:submitted workload)))))
-
 (deftest test-append-to-aou-not-started
   (with-redefs-fn {#'aou/submit-aou-workflow mock-submit-workload}
-    #(is
-      (thrown? Exception
-               (workloads/append-to-workload! [workloads/aou-sample]
-                                              (workloads/create-workload! (make-aou-workload-request)))))))
+    #(let [workload (workloads/create-workload! (make-aou-workload-request))]
+       (is (thrown? Exception (workloads/append-to-workload!
+                               [workloads/aou-sample]
+                               workload))))))
+
+(deftest test-append-to-stopped-aou-workload
+  (with-redefs-fn {#'aou/submit-aou-workflow mock-submit-workload}
+    #(as-> (workloads/create-workload! (make-aou-workload-request)) workload
+       (workloads/start-workload! workload)
+       (workloads/stop-workload! workload)
+       (is (thrown? Exception (workloads/append-to-workload!
+                               [workloads/aou-sample]
+                               workload))))))
+
+(deftest test-workload-state-transition
+  (as-> (make-aou-workload-request) $
+    (doto (workloads/create-workload! $)
+      (-> (contains? :created)  is)
+      (-> (absent?   :started)  is)
+      (-> (absent?   :stopped)  is)
+      (-> (absent?   :finished) is))
+    (doto (workloads/update-workload! $)
+      (-> (contains? :created)  is)
+      (-> (absent?   :started)  is)
+      (-> (absent?   :stopped)  is)
+      (-> (absent?   :finished) is))
+    (doto (workloads/start-workload! $)
+      (-> (contains? :created)  is)
+      (-> (contains? :started)  is)
+      (-> (absent?   :stopped)  is)
+      (-> (absent?   :finished) is))
+    (doto (workloads/stop-workload! $)
+      (-> (contains? :created)  is)
+      (-> (contains? :started)  is)
+      (-> (contains? :stopped)  is)
+      (-> (absent?   :finished) is))
+    (doto (workloads/update-workload! $)
+      (-> (contains? :created)  is)
+      (-> (contains? :started)  is)
+      (-> (contains? :stopped)  is)
+      (-> (contains? :finished) is))))
+
+(deftest test-stop-workload-state-transition
+  (as-> (make-aou-workload-request) $
+    (doto (workloads/create-workload! $)
+      (-> (contains? :created)  is)
+      (-> (absent?   :started)  is)
+      (-> (absent?   :stopped)  is)
+      (-> (absent?   :finished) is))
+    (doto (workloads/stop-workload! $)
+      (-> (contains? :created)  is)
+      (-> (contains? :started)  is)
+      (-> (contains? :stopped)  is)
+      (-> (contains? :finished) is))))
```

Comment:
Clever, Yoda.  Why `$` instead of `request`?
Why not a table of expected states?
(Maybe this is the expected refactoring you mentioned?)

Reply:
Ah yeah. I personally don't like the yoda but not sure how to get around it. I used `$` as 
1. It's not always a request, it's also a workload. I'd like to eliminate the symbol altogether tbh
2. The clojure doc uses `$` so why not

Reply:
The refactoring is because it's in every module integ test. They're all fairly similar tbh which reflects the state of source code.